CMakeLists.txt

```diff
@@ -130,8 +137,7 @@ if(OpenCL_FOUND)
         add_library(Compute INTERFACE)
         add_library(VexCL::Compute ALIAS Compute)
 
-        target_include_directories(Compute INTERFACE ${OpenCL_INCLUDE_DIRS})
-        target_link_libraries(Compute INTERFACE Common ${OpenCL_LIBRARIES})
+        target_link_libraries(Compute INTERFACE Common OpenCL::OpenCL)
```

Comment:
Does this need both `Common` and `OpenCL::OpenCL`?

Reply:
Or, rather, does `Common` need `OpenCL::OpenCL` in line 131 above?

Reply:
Sorry, my head is not working right today. Was looking at the wrong argument.

Reply:
Yes; or it could have just `OpenCL`, which already have `Common` and `OpenCL::OpenCL`.

There is a downside with using names like `Common` and `OpenCL`; if this is `add_subproject`'d into another project, that project cannot have any targets named similarly. It might make sense to apply the same fix I've added to Minuit2, where the internal names are something like VexOpenCL, and then `ALIAS` (`add_subproject`) and IMPORTED INTERFACE (Config) targets provide the pretty namespaced names to users. CMake is missing a "Namespace" option to `add_subdirectory`. (That's an idea for later, not here)

Reply:
(The confusion between OpenCL (VexCL) and OpenCL::OpenCL (OpenCL) reminded me of that)